omod/src/main/webapp/resources/scripts/app.js

```diff
@@ -64,8 +65,9 @@ app.controller('tagCtrl',['$scope','TagService','ngDialog',
     $scope.navigate = function (tag) {
         emr.navigateTo({
             provider: 'tag',
+
             page: 'taggedObjects',
-            query: { tag : tag.display }
+            query: { tag : tag.display, returnUrl: $scope.returnUrl }
```

Comment:
Why not just pass the patient uuid and construct the url on demand on the tagged objects page?

Reply:
Ignore my comment above, I now know why you did this way.